task-ballerina/scheduler.bal

```diff
@@ -14,98 +14,156 @@
+# + return - A `task:Error` if the process failed due to any reason or else ()
+public isolated function configureWorkerPool(int workerCount = 5, time:Seconds waitingTime = 5)
+                                returns Error? {
+    return configureThread(workerCount, <int>(waitingTime * 1000.0));
+}
 
-    # Detaches the provided `service` from the task.
-    #
-    # + attachedService - Ballerina `service` object, which needs to be detached from the task
-    # + return - A `task:SchedulerError` if the process failed due to any reason or else ()
-    public isolated function detach(service object {} attachedService) returns SchedulerError? {
-        var result = detachExternal(self.taskListener, attachedService);
-        if (result is ListenerError) {
-            string message = "Scheduler failed to detach the service";
-            return error SchedulerError(message, result);
-        }
-    }
+# Schedule the given `job` for the given time. Once scheduled, it will return a job ID, which can be used to manage
+# the job.
+#
+# + triggerTime - The specific time in Ballerina `time:Civil` to trigger only one time
+# + job - Ballerina job, which is to be executed during the trigger.
+# + return - A job ID or else a `task:Error` if the process failed due to any reason
+public isolated function scheduleOneTimeJob(Job job, time:Civil triggerTime) returns JobId|Error {
+    int result = check scheduleJob(job, check getTimeInMillies(triggerTime));
+    JobId jobId = {id: result};
+    return jobId;
+}
 
-    # Starts running the task. Task Scheduler will not run until this has been called.
-    #
-    # + return - A `task:SchedulerError` if the process failed due to any reason or else ()
-    public isolated function 'start() returns SchedulerError? {
-        var result = startExternal(self.taskListener);
-        if (result is ListenerError) {
-            string message = "Scheduler failed to start";
-            return error SchedulerError(message, result);
-        }
+# Schedule the recurring `job` according to the given duration. Once scheduled it will return the job ID which
+# can be used to manage the job.
+#
+# + job - Ballerina job, which is to be executed by the scheduler.
+# + interval - The duration of the trigger (in seconds), which is used to run the job frequently
+# + maxCount - The maximum number of trigger counts
+# + startTime - The trigger start time in Ballerina `time:Civil`. If it is not provided, a trigger will
+#               start immediately
+# + endTime - The trigger end time in Ballerina `time:Civil`
+# + taskPolicy -  The policy, which is used to handle the error and will be waiting during the trigger time
+# + return - A job ID or else a `task:Error` if the process failed due to any reason
+public isolated function scheduleJobRecurByFrequency(Job job,  decimal interval,  int maxCount = -1,
+                                    time:Civil? startTime = (), time:Civil? endTime = (), TaskPolicy taskPolicy = {})
```

Comment:
Why do we use `time:Civil` here, instead of `time:Utc`?. I think the scheduling time of a job should independent from a timezone.

Reply:
I thought the User can define the time by using `time:Civil` easier than `time:Utc` and also set the time zone in that. 
I am internally converting this to [time:UTc](https://github.com/ballerina-platform/module-ballerina-task/pull/77/files#diff-bc2bf356bac8ea867931ee7cc2e119dd3b180d5d0df7d076f012ce1d92f22d84R65). So, Can we go this way?